src/components/Planning/GroupForm/GroupProjectSearch.tsx

```diff
@@ -0,0 +1,155 @@
+  const [value, setValue] = useState('');
+  const [searchedProjects, setSearchedProjects] = useState<Array<IOption>>([]);
+  const [queryParams, setQueryParams] = useState<IQueryParamsState>({
+    masterClass: '',
+    class: '',
+    subClass: '',
+    district: '',
+    division: '',
+    subDivision: '',
+  });
+
+  useEffect(() => {
+    console.log('firing');
+    setQueryParams((current) => ({
+      ...current,
+      masterClass: getValues().masterClass?.value ? getValues().masterClass?.value : '',
+      class: getValues().class?.value ? getValues().class?.value : '',
+      subClass: getValues().subClass?.value ? getValues().subClass?.value : '',
+      district: getValues().district?.value ? getValues().district?.value : '',
+      division: getValues().division?.value ? getValues().division?.value : '',
+      subDivision: getValues().subDivision?.value ? getValues().subDivision?.value : '',
+    }));
+  }, [getValues]);
+  const handleValueChange = useCallback((value: string) => setValue(value), []);
+
+  const getSuggestions = useCallback(
+    (inputValue: string) =>
+      new Promise<{ value: string; label: string }[]>((resolve, reject) => {
+        // printing out search params for later
+        const queryString = buildQueryParamString(queryParams, inputValue);
+        console.log(queryString);
+
+        getProjectsWithParams(queryString)
+          .then((res) => {
+            if (res) {
+              // Filter out only the projects which haven't yet been added to be the submitted list from the result
+              const resultList = res.projects?.filter(
+                ({ project }) =>
+                  !arrayHasValue(
+                    projectsForSubmit.map((p) => p.value),
+                    project.id,
+                  ),
+              );
+              console.log(resultList);
+
+              // Convert the resultList to options for the suggestion dropdown
+              const searchProjectsItemList: Array<IOption> | [] = resultList
+                ? resultList.map(({ project }) => ({
+                    ...listItemToOption({ id: project.id, value: project.name }),
+                  }))
+                : [];
+              if (searchProjectsItemList.length > 0) {
+                setSearchedProjects(searchProjectsItemList);
+              }
+
+              resolve(searchProjectsItemList);
+            }
+            resolve([]);
+          })
+          .catch(() => reject([]));
+      }),
+    [projectsForSubmit, getValues],
+  );
+
+  const handleSubmit = (value: string) => {
+    value && onProjectClick(searchedProjects.find((p) => p.label === value));
+    setValue('');
+  };
+
+  const handleDelete = (e: React.MouseEvent<HTMLButtonElement>) => {
+    onProjectSelectionDelete(
+      (e.currentTarget as HTMLButtonElement)?.parentElement?.innerText as string,
+    );
+  };
+  return (
+    <div className="dialog-section" data-testid="search-project-field-section">
```

Comment:
SearchInput should be wrapped in a <Controller name="projectsForSubmit" />

Reply:
Done